interfaces/ifacetest/testiface.go

```diff
@@ -44,6 +44,13 @@ type TestInterface struct {
 	PlugSnippetCallback func(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error)
 	// PermanentPlugSnippetCallback is the callback invoked inside PermanentPlugSnippet()
 	PermanentPlugSnippetCallback func(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error)
+
+	// Support for interacting with the test backend.
+
+	RecordTestConnectedPlugCallback func(rec *TestRecorder, plug *interfaces.Plug, slot *interfaces.Slot) error
```

Comment:
On the interfaces themselves, I think we can have simply, e.g., `ApparmorConnectedPlug`. No need to have `Define` or `Record` on them.